tensorflow/compiler/mlir/tosa/transforms/convert_tfl_uint8.cc

```diff
@@ -130,6 +130,63 @@ struct ConvertUint8QConstOp : public RewritePattern {
   }
 };
 
+namespace {
+
+// returns true iff @a type is a shaped type with element type that is uint8
+// if it is, then return the rescaled type, uint8_zp, and output_zp to use to
+// rescale type to signed type with adjusted zero point.
+bool IsShapedUint8Type(OpBuilder& builder, const Type type, Type& rescaled_type,
+                       int32_t& uint8_zp, int32_t& output_zp) {
+  auto uint8_type = type.dyn_cast<mlir::ShapedType>();
+  if (!uint8_type) return false;
+
+  auto element_type = uint8_type.getElementType();
+  auto uint8_element_quant_type =
+      element_type.dyn_cast<mlir::quant::UniformQuantizedType>();
+  bool is_uint8_element_quant_type =
+      uint8_element_quant_type && !uint8_element_quant_type.isSigned() &&
+      uint8_element_quant_type.getStorageTypeIntegralWidth() == 8;
+  bool is_uint8_element_type = element_type.isUnsignedInteger(8);
+  if (!is_uint8_element_quant_type && !is_uint8_element_type) return false;
+
+  // type has uint8 element type
+  if (is_uint8_element_quant_type) {
```

Comment:
Rather than getting two bools and bailing, you  should do something like:

```
if (auto quantType = dyn_cast<mlir::quant::UniformQuantizedType>()) {
  if (quantType.isSigned() || quantType.getStorageTypeIntegralWidth() != 8) return false;
  .....
  return true;
}

// Repeat for dyn_cast<IntegerType>();
if (...) {
  ...
  return true;
}

return false;
```

Reply:
fixed